### PR TITLE
Fix for #2184: Add space after separator

### PR DESCRIPTION
**Pull request by matthiasgeiger**
... seems to be as simple as that.

Change has been introduced as the delimitor is no longer saved as a String ", " but as a `Character` which chops away the space after the comma.

- ~~[ ] Change in CHANGELOG.md described~~ introduced in dev Builds
- [x] Tests created for changes
- ~~[ ] Screenshots added (for bigger UI changes)~~
- [x] Manually tested changed features in running JabRef
- ~~[ ] Check documentation status (Issue created for outdated help page at [help.jabref.org](https://github.com/JabRef/help.jabref.org/issues)?)~~
- ~~[ ] If you changed the localization: Did you run `gradle localizationUpdate`?~~

### CODE REVIEW

**Comment by lenhard** [comment]
That looks simple indeed. 

Since we cannot let you get away with a one-liner: Please add a test that detects the bug you fixed, so that the build would break if this happens again.

**Comment by tobiasdiez** [comment]
Yes this fix works. I'm unsure if we should use the KeywordList class here (since in the end we assume that the field is a list of words and we are adding a new keyword).

**Comment by matthiasgeiger** [comment]
@tobiasdiez Sorry I don't get it. I've put it in the `KeywordGroup` class as the `add(List<BibEntry> entriesToAdd)` method is not overriden in the subclass `ExplicitGroup`.

Apart from that: If talking only about keywords (without the group context) the keywordSeparator is there followed by a `+" "`, too. (Sorry, I'm currently on my way home and have just closed IntelliJ - so I cannot provide a pointer for now.)

**Comment by tobiasdiez** [comment]
@matthiasgeiger sorry for being cryptic. What I meant was that the class [KeywordList](https://github.com/JabRef/jabref/blob/955b94d7221e5ee3c102731f885eb395e1c2ad64/src/main/java/net/sf/jabref/model/entry/KeywordList.java) probably should be used to modify the field (instead of being text-based).
The code then looks similar to:
````
String oldContent = ...
KeywordList wordlist = new KeywordList(oldContent);
wordlist.add(searchExpression);
String newContent = wordlist.getAsString(seperator);
````

**Comment by matthiasgeiger** [comment]
Ahhh, I see... and this is exactly the position where the `+ "  "` addition is made for keywords.

Will check whether using this is actually possible without sideeffects.

File: src/test/java/net/sf/jabref/model/groups/ExplicitGroupTest.java [hunk 1]
```diff
@@ -20,17 +21,36 @@ public void testToStringSimple() {
...
+    public void testEntryStorageSingleGroups() {
+        BibEntry entry = new BibEntry();
+        ExplicitGroup group = new ExplicitGroup("myExplicitGroup", GroupHierarchyType.INDEPENDENT, ',');
+        group.add(entry);
+        assertEquals(Optional.of("myExplicitGroup"), entry.getField(FieldName.GROUPS));
+    }
+
+    @Test
+    public void testEntryStorageTwoGroups() {
+        BibEntry entry = new BibEntry();
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/model/groups/ExplicitGroupTest.java]: please extract all the initialization (entry, group, group2) to a setup method and reuse it in the tests.

File: src/test/java/net/sf/jabref/model/groups/ExplicitGroupTest.java [hunk 2]
```diff
@@ -20,17 +21,36 @@ public void testToStringSimple() {
...
+        ExplicitGroup group = new ExplicitGroup("myExplicitGroup", GroupHierarchyType.INDEPENDENT, ',');
+        ExplicitGroup group2 = new ExplicitGroup("myExplicitGroup2", GroupHierarchyType.INDEPENDENT, ',');
+        group.add(entry);
+        group2.add(entry);
+        assertEquals(Optional.of("myExplicitGroup, myExplicitGroup2"), entry.getField(FieldName.GROUPS));
+    }
+
+    @Test
+    public void testGroupStorageAlreadyInGroup() throws Exception {
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/model/groups/ExplicitGroupTest.java]: please remove `test` prefix from test names and add the expected outcome to the name (e.g. addToAlreadyPresentGroupDoesNotChangeEntry)

File: src/test/java/net/sf/jabref/model/groups/KeywordGroupTest.java [hunk 3]
```diff
@@ -85,4 +88,37 @@ public void containsWordWithWhitespaceInSentence() throws Exception {
 
         assertTrue(group.isMatch(entry));
     }
+
+    @Test
+    public void testGroupStorageEmptyKeywords() throws Exception {
+        KeywordGroup group = new KeywordGroup("name", FieldName.KEYWORDS, "test", false, false, GroupHierarchyType.INDEPENDENT,
```
Comment (tobiasdiez) [inline src/test/java/net/sf/jabref/model/groups/KeywordGroupTest.java]: same remarks as above: setup() and test names.

**Comment by matthiasgeiger** [comment]
Done. Will merge after Travis is green.

**Pull request merged**